### ISSUE TITLE
Collection becomes unavailable mid-write

### ISSUE THREAD

**Issue by jcjimenez** [Azure/azure-cosmosdb-spark#103]
I'm seeing an error where writing a set of about 5k records to cosmos gives the error below. Could this because of some form of throttling? (happy to include sample code)

```
Caused by: com.microsoft.azure.documentdb.DocumentClientException: {"Errors":["The requested resource is no longer available at the server."]}
ActivityId: 8381ac79-121a-4b1e-a759-eb2ea53fac4b, Request URI: /apps/1921ec42-1c88-4fdd-8d6c-78646cba4b15/services/44b332e9-d210-480d-b322-bc7d9de32b23/partitions/130361cc-a639-470d-b64f-ea88a5c3679a/replicas/131474622017806820p//, StatusCode: Gone
	at com.microsoft.azure.documentdb.internal.ErrorUtils.maybeThrowException(ErrorUtils.java:69)
	at com.microsoft.azure.documentdb.internal.directconnectivity.HttpTransportClient.processResponse(HttpTransportClient.java:151)
	at com.microsoft.azure.documentdb.internal.directconnectivity.HttpTransportClient.invokeStore(HttpTransportClient.java:121)
	at com.microsoft.azure.documentdb.internal.directconnectivity.HttpTransportClient.invokeStore(HttpTransportClient.java:128)
	at com.microsoft.azure.documentdb.internal.directconnectivity.TransportClient.invokeResourceOperation(TransportClient.java:12)
	at com.microsoft.azure.documentdb.internal.directconnectivity.ReplicatedResourceClient.write(ReplicatedResourceClient.java:76)
	at com.microsoft.azure.documentdb.internal.directconnectivity.ReplicatedResourceClient.invoke(ReplicatedResourceClient.java:51)
	at com.microsoft.azure.documentdb.internal.directconnectivity.ServerStoreModel$1.apply(ServerStoreModel.java:84)
	at com.microsoft.azure.documentdb.internal.RetryUtility.executeStoreClientRequest(RetryUtility.java:113)
	at com.microsoft.azure.documentdb.internal.directconnectivity.ServerStoreModel.processMessage(ServerStoreModel.java:89)
	at com.microsoft.azure.documentdb.DocumentClient$4.apply(DocumentClient.java:2900)
	at com.microsoft.azure.documentdb.internal.RetryUtility.executeDocumentClientRequest(RetryUtility.java:58)
	at com.microsoft.azure.documentdb.DocumentClient.doCreate(DocumentClient.java:2905)
	at com.microsoft.azure.documentdb.DocumentClient.access$100(DocumentClient.java:88)
	at com.microsoft.azure.documentdb.DocumentClient$1.apply(DocumentClient.java:761)
	at com.microsoft.azure.documentdb.internal.RetryUtility.executeCreateDocument(RetryUtility.java:163)
	at com.microsoft.azure.documentdb.DocumentClient.createDocument(DocumentClient.java:766)
	at com.microsoft.azure.documentdb.rx.internal.RxWrapperDocumentClientImpl$17.invoke(RxWrapperDocumentClientImpl.java:387)
	at com.microsoft.azure.documentdb.rx.internal.RxWrapperDocumentClientImpl$17.invoke(RxWrapperDocumentClientImpl.java:384)
	at com.microsoft.azure.documentdb.rx.internal.RxWrapperDocumentClientImpl$2.call(RxWrapperDocumentClientImpl.java:151)
	at com.microsoft.azure.documentdb.rx.internal.RxWrapperDocumentClientImpl$2.call(RxWrapperDocumentClientImpl.java:145)
```

**Comment by jcjimenez** [comment]
It's worth noting that some records do make it into the table.

**Comment by dennyglee** [comment]
Yes, this is most likely due to throttling.  Could you provide a sample so we can do a repro?  Thanks!

**Comment by khdang** [comment]
Does this happen a lot to you? 
Does it succeed if you retry? If yes it could be because of partition movement and the SDK could possibly have an issue handling it. I assume you're not doing any other operations on the same collection.

**Comment by jcjimenez** [comment]
@khdang @dennyglee yeah, it was happening because:

1. I had created my collection with something like a value of 4000 RU/s
2. I had not set any values for QueryMaxRetryOnThrottled or QueryMaxRetryWaitTimeSecs.

After adjusting both 1 and 2, I was able to upload without any problems. This is the code I was using: https://github.com/Azure/azure-cosmosdb-spark/pull/110/files#diff-f47cb359966160a799ab6793501aaca4

So, if we are already changing the exception message (or decided against it or for a different approach), I am in favor of closing this.
